.cascade-code/Chapman.edu/_components/omni_nav_v2/omni_nav.vtl

```diff
@@ -112,7 +112,14 @@
     ## Is current page directory $pagePathDir under this $navDirectory? If so,
     ## we found our $navDirectory.
     #if ( $pagePath.contains($dirName) && $pagePathDir == $dirName )
-      #set ( $pageUmbrellaCategory = $umbrellaCategory )
+      ## Hasn't been set yet in a previous iteration
+      #if ( $pageUmbrellaCategory.size() == 0 )
+        #set ( $pageUmbrellaCategory = $umbrellaCategory )
+      ## If it has been set already and this is another match,
+      ## check if this new one is longer (more specific directory)
```

Comment:
The comment is restarting the code rather than explaining why. A why comment would be helpful here.